packages/components/src/progress-bar/index.tsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import type { ProgressBarProps } from './types';
+import type { WordPressComponentProps } from '../ui/context';
+
+// Width of the indicator for the indeterminate progress bar
+export const INDETERMINATE_TRACK_WIDTH = 50;
+
+export function ProgressBar(
+	props: WordPressComponentProps< ProgressBarProps, 'div', false >
+) {
+	const { className, value, indicatorColor, trackColor } = props;
+	const isIndeterminate = ! Number.isFinite( value );
+	const wrapperClasses = classnames( 'components-progress-bar', className, {
+		'is-indeterminate': isIndeterminate,
+	} );
+	const trackStyle = {
+		backgroundColor: trackColor ? trackColor : undefined,
```

Comment:
Is this check necessary? If `trackColor` can only ever be a colour or `undefined` (based on its type), can we not just assign `trackColor` to `backgroundColor`?

Similarly for `indicatorColor` below.

Reply:
Good point, but this is no longer relevant since I'm going to utilize the Theme system for these instead, as @mirka suggested in [this comment](https://github.com/WordPress/gutenberg/pull/53030#discussion_r1276950940).